### PR TITLE
Expose additional GlobalDNSProvider params

### PR DESCRIPTION
**Pull request by prachidamle**
This is a PR to add additional parameters to configure external-dns provider
https://github.com/rancher/rancher/issues/19251

### CODE REVIEW

File: apis/management.cattle.io/v3/globaldns_types.go [hunk 1]
```diff
@@ -52,13 +52,18 @@ type GlobalDNSProviderSpec struct {
 }
 
 type Route53ProviderConfig struct {
-	AccessKey string `json:"accessKey" norman:"notnullable,required,minLength=1"`
-	SecretKey string `json:"secretKey" norman:"notnullable,required,minLength=1,type=password"`
+	AccessKey       string `json:"accessKey" norman:"notnullable,required,minLength=1"`
+	SecretKey       string `json:"secretKey" norman:"notnullable,required,minLength=1,type=password"`
+	CredentialsPath string `json:"credentialsPath" norman:"default=/.aws"`
+	RoleArn         string `json:"roleArn"`
```
Comment (alena1108) [inline apis/management.cattle.io/v3/globaldns_types.go]: shall we add omitempty to all the new values?

File: client/management/v3/zz_generated_cloudflare_provider_config.go [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+	CloudflareProviderConfigFieldAPIKey       = "apiKey"
+	CloudflareProviderConfigFieldProxySetting = "proxySetting"
 )
 
 type CloudflareProviderConfig struct {
-	APIEmail string `json:"apiEmail,omitempty" yaml:"apiEmail,omitempty"`
-	APIKey   string `json:"apiKey,omitempty" yaml:"apiKey,omitempty"`
+	APIEmail     string `json:"apiEmail,omitempty" yaml:"apiEmail,omitempty"`
+	APIKey       string `json:"apiKey,omitempty" yaml:"apiKey,omitempty"`
+	ProxySetting string `json:"proxySetting,omitempty" yaml:"proxySetting,omitempty"`
```
Comment (alena1108) [inline client/management/v3/zz_generated_cloudflare_provider_config.go]: do we only want proxy parameter for cloudFlare? I recall seeing more params via answers, are any of them useful enough to be exposed?

Comment (prachidamle) [reply client/management/v3/zz_generated_cloudflare_provider_config.go]: No, rest of the parameters are not for configuring cloudflare functionality. They are mainly related to the external-dns pod deployment (use k8s rbac, ownerId to know which records are created by this pod, do a full-sync Vs only add new dns records etc.)

File: apis/management.cattle.io/v3/globaldns_types.go (hunk 1 above)
Comment (prachidamle) [reply apis/management.cattle.io/v3/globaldns_types.go]: I think we dont need omitempty for the ones which have a default specified. For the property "RoleArn" I will add omitempty.

**Comment by alena1108** [comment]
LGTM

**Pull request merged**